metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -1198,15 +1198,22 @@ class MetalsLspService(
   }
 
   def didChangeWatchedFiles(
-      paths: List[AbsolutePath]
+      events: List[FileEvent]
   ): Future[Unit] = {
-    val importantPath =
-      paths
-        .filterNot(path =>
-          savedFiles.isRecentlyActive(path) || path.isDirectory
+    val importantEvents =
+      events
+        .filterNot(event =>
+          event.getUri().toAbsolutePathSafe match {
+            case None => true
+            case Some(path) =>
+              savedFiles.isRecentlyActive(path) || path.isDirectory
+          }
         ) // de-duplicate didSave events.
         .toSeq
-    onChange(importantPath)
+    val (deleteEvents, changeAndCreateEvents) =
+      importantEvents.partition(_.getType().equals(FileChangeType.Deleted))
+    deleteEvents.map(_.getUri().toAbsolutePath).foreach(onDelete)
+    onChange(changeAndCreateEvents.map(_.getUri().toAbsolutePath))
```

Comment:
Should I maybe call the other `didChangeWatchedFiles` function here?

Reply:
Hmm... the events are totally different, but possibly it would be great to deduplicate the logic. 

But it's also fine as is.

Reply:
I can map those events to the other ones. See the newest commit.